pop.go

```diff
@@ -18,13 +18,47 @@ import (
 	"crypto/sha1"
 	"encoding/base64"
 	"fmt"
+	"io/ioutil"
+	"net/http"
 	"net/url"
 	"sort"
 	"strings"
+	"time"
+	//	"github.com/google/uuid"
 )
 
+// reuse http client
+type popClient struct {
+	*http.Client
+}
+
+func newPOP(timeout time.Duration) *popClient {
+	if timeout < 0 {
+		timeout = 10 * time.Second
+	}
+	return &popClient{
+		&http.Client{
+			Timeout: timeout,
+		},
+	}
+}
+
+func (pop *popClient) request(params map[string]string, url, accessSecret string) (string, error) {
+	sign := pop.signature(params, "POST", accessSecret)
+	params["Signature"] = sign
+
+	rsp, err := pop.PostForm(url, convertMap2URLValues(params))
+	if err != nil {
+		return "", err
+	}
+	defer rsp.Body.Close()
+
+	body, _ := ioutil.ReadAll(rsp.Body)
```

Comment:
Please handle the returned error value.

Reply:
Thanks.
Fixed in [ref](https://github.com/sql-machine-learning/goalisa/pull/12/files#diff-3ef3327702394ae04f1ba6329e0feb52L56-L57)